qiskit_nature/second_q/formats/fcidump/fcidump.py

```diff
@@ -0,0 +1,131 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""FCIDump"""
+
+from __future__ import annotations
+
+from typing import List
+from dataclasses import dataclass
+from pathlib import Path
+import numpy as np
+
+from qiskit_nature import QiskitNatureError
+
+
+from .dumper import _dump_1e_ints, _dump_2e_ints, _write_to_outfile
+from .parser import _parse
+
+
+@dataclass
+class FCIDump:
+    """
+    Qiskit Nature dataclass for reading an FCIDump file.
+
+    The FCIDump format is partially defined in Knowles1989.
+
+    References:
+        Knowles1989: Peter J. Knowles, Nicholas C. Handy,
+            A determinant based full configuration interaction program,
+            Computer Physics Communications, Volume 54, Issue 1, 1989, Pages 75-83,
+            ISSN 0010-4655, https://doi.org/10.1016/0010-4655(89)90033-7.
+    """
+
+    hij: np.ndarray | None
```

Comment:
```suggestion
    hij: np.ndarray
```
I'm actually thinking now, that this should be a required argument